src/consumer/stream_consumer.rs

```diff
@@ -155,15 +155,17 @@ impl<C: ConsumerContext> FromClientConfigAndContext<C> for StreamConsumer<C> {
         let stream_consumer = StreamConsumer {
             consumer: Arc::new(BaseConsumer::from_config_and_context(config, context)?),
             should_stop: Arc::new(AtomicBool::new(false)),
-            handle: Mutex::new(None),
         };
         Ok(stream_consumer)
     }
 }
 
 impl<C: ConsumerContext> StreamConsumer<C> {
-    /// Starts the StreamConsumer with default configuration (100ms polling interval and no
-    /// `NoMessageReceived` notifications).
+    /// Starts the StreamConsumer with default configuration (100ms polling
+    /// interval and no `NoMessageReceived` notifications).
+    ///
+    /// **Note:** this method must be called from within the context of a Tokio
```

Comment:
maybe provide

```
pub fn start_with_timer<F>(...., timerFn: F) -> MessageStream<C>
   where F: Fn() -> Future
```

And make `start_with` conditional on tokio feature?

Reply:
I don't think the `time` feature is particularly onerous to uncondtionally require. It only introduces a dependency on `slab` which is quite lightweight.

Reply:
I dont agree, it introduces dependency on tokio. Previously rust-rdkafka crate could be used with async-std or smol runtimes.